changelog/assert_ctfe.dd

```diff
@@ -0,0 +1,20 @@
+ctfe-only function elision
+
+As of now assert(__ctfe) inside the outmoster scope of a function body or inside an in contract will cause the function to not emited into the object file.
+Should a another function attempt to use a ctfe-only function a compile error will be emited.
+
+-------
+string generateMixin(string name)
+{
+    assert(__ctfe);
+    return "\"This is not a good mixin\"";
```

Comment:
I don't think we shouldn't be encouraging users to write code this when:

```d
return `"This is not a good mixin"`;
```
is far more readable.

Reply:
Is it deliberate for demo purposes that the input parameter is not actually used? :-)

Reply:
oh no ... it's not thanks for alerting me to it.